code/game/machinery/hologram.dm

```diff
@@ -68,197 +105,382 @@ var/list/holopads = list()
+/obj/machinery/hologram/holopad/proc/transfer_to_nearby_pad(turf/T, mob/holo_owner)
+	for(var/pad in holopads)
+		var/obj/machinery/hologram/holopad/another = pad
+		if(another == src)
+			continue
+		if(another.validate_location(T))
+			var/obj/effect/overlay/holo_pad_hologram/h = masters[holo_owner]
+			unset_holo(holo_owner)
+			another.set_holo(holo_owner, h)
+			return TRUE
+	return FALSE
+
+/obj/machinery/hologram/holopad/proc/validate_user(mob/living/user)
+	if(QDELETED(user) || user.incapacitated() || !user.client)
+		return FALSE
+
+	if(istype(user, /mob/living/silicon/ai))
+		var/mob/living/silicon/ai/AI = user
+		if(!AI.current)
+			return FALSE
+	return TRUE
+
+//Can we display holos there
+//Area check instead of line of sight check because this is a called a lot if AI wants to move around.
+/obj/machinery/hologram/holopad/proc/validate_location(turf/T,check_los = FALSE)
+	if(T.z == z && get_dist(T, src) <= holo_range && T.loc == get_area(src))
+		return TRUE
 	else
+		return FALSE
+
+
+/obj/machinery/hologram/holopad/proc/move_hologram(mob/living/user, turf/new_turf)
+	if(masters[user])
+		var/obj/effect/overlay/holo_pad_hologram/holo = masters[user]
+		var/transfered = FALSE
+		if(!validate_location(new_turf))
+			if(!transfer_to_nearby_pad(new_turf,user))
+				clear_holo(user)
+				return FALSE
+			else
+				transfered = TRUE
+		//All is good.
+		holo.setDir(get_dir(holo.loc, new_turf))
+		holo.forceMove(new_turf)
+		if(!transfered)
+			update_holoray(user,new_turf)
+	return TRUE
+
+/obj/machinery/hologram/holopad/proc/activate_holo(mob/living/user, var/force = 0)
+	var/mob/living/silicon/ai/AI = user
+	if(!istype(AI))
+		AI = null
+	if(AI && !force && AI.eyeobj.loc != loc) // allows holopads to pass off holograms to the next holopad in the chain
 		to_chat(user, "<font color='red'>ERROR:</font> Unable to project hologram.")
-	return
+	if(!(stat & NOPOWER) && (!AI || force))
+		if(AI && (istype(AI.current, /obj/machinery/hologram/holopad)))
+			to_chat(user, "<span class='danger'>ERROR:</span> Image feed in progress.")
+			return
+
+		var/obj/effect/overlay/holo_pad_hologram/hologram = new(loc)//Spawn a blank effect at the location.
+		if(isAI(user))
+			hologram.icon = AI.holo_icon
+		else	//make it like real life
+			hologram.icon = getHologramIcon(get_id_photo(user))
+			hologram.icon_state = user.icon_state
+			hologram.alpha = 100
+			hologram.Impersonation = user
+
+		hologram.mouse_opacity = 0//So you can't click on it.
+		hologram.layer = FLY_LAYER//Above all the other objects/mobs. Or the vast majority of them.
+		hologram.anchored = 1//So space wind cannot drag it.
+		hologram.name = "[user.name] (hologram)"//If someone decides to right click.
+		hologram.set_light(2)	//hologram lighting
+		move_hologram()
+
+		set_holo(user, hologram)
+
+		if(!masters[user])//If there is not already a hologram.
+			visible_message("A holographic image of [user] flicks to life right before your eyes!")
+
+		return hologram
+
+	else

```

Comment:
Tay is correct, just remove this `else` and untab the code. The above block always returns early.